src/containers/quiz/quiz-header/QuizHeader.tsx

```diff
@@ -33,22 +46,39 @@ const QuizHeader = ({
   isGraded,
   questionsAnswered,
   createdAt,
-  updatedAt
+  updatedAt,
+  isNotStarted,
+  quizItems,
+  settings,
+  handlePreview
 }: QuizHeaderProps) => {
   const { userRole } = useAppSelector((state) => state.appMain)
-
   const isStudent = userRole === UserRoleEnum.Student
 
   const isTutor = userRole === UserRoleEnum.Tutor
+  const { id, quizId } = useParams()
```

Comment:
Add this default values here, and you won't use condition in `useCallback`
```suggestion
  const { id = '', quizId = '' } = useParams()
```